frontend/src/visualization/filtering/FilterLineProperty.tsx

```diff
@@ -7,23 +7,59 @@ import {
   MenuItem,
   Select,
 } from '@material-ui/core';
-import React from 'react';
-import FilterPropertyModel from '../../FilterPropertyModel';
+import React, { useEffect } from 'react';
+import useService from '../../dependency-injection/useService';
+import FilterStateStore from '../../stores/filterState/FilterStateStore';
+import { FilterModelEntry } from '../../shared/filter';
 
 const useStyles = makeStyles(() =>
   createStyles({
     select: {
-      width: 200,
+      width: 500,
     },
   })
 );
 
-const EntityPropertySelect = (props: {
-  property: FilterPropertyModel;
-  setProperty: React.Dispatch<React.SetStateAction<FilterPropertyModel>>;
+const FilterLineProperty = (props: {
+  filterModelEntry: FilterModelEntry;
+  filterLineType: string;
+  entity: 'node' | 'edge';
+  applyFilter: boolean;
+  setApplyFilter: React.Dispatch<React.SetStateAction<boolean>>;
 }): JSX.Element => {
   const classes = useStyles();
-  const { property, setProperty } = props;
+  const {
+    filterModelEntry,
+    filterLineType,
+    entity,
+    applyFilter,
+    setApplyFilter,
+  } = props;
+
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
+
+  const [selectedValues, setSelectedValues] = React.useState<string[]>(
+    filterStateStore.getPropertyStateValues(
+      filterLineType,
+      filterModelEntry.key,
+      entity
+    ) ?? []
+  );
+
+  // update filterStateStore here because selectedValues will first be updated in the next render
+  useEffect(() => {
+    if (applyFilter) {
+      filterStateStore.addFilterPropertyState(
+        {
+          name: filterModelEntry.key,
+          values: selectedValues,
+        },
+        filterLineType,
+        entity
+      );
+      setApplyFilter(false);
+    }
+  }, [selectedValues, applyFilter]);
```

Comment:
I don't really understand why `setApplyFilter(false)` is called here and why `selectedValues` is in the depenecy list. Looks like a hack, please add a description.

Reply:
For `applyFilter`: The store only needs to update if the "Apply filter" button is pressed. The "Apply filter" button is a component of the parent. `applyFilter` signals that the button is pressed -> The store can be updated. Otherwise, if the filter is canceled (Cancel button pressed) the store shall not be updated. 

For `selectedValues`: If new values are selected the store needs to update. So the store first updates when all values are selected and not a single one.

Reply:
> For applyFilter: The store only needs to update if the "Apply filter" button is pressed. The "Apply filter" button is a component of the parent. applyFilter signals that the button is pressed -> The store can be updated. Otherwise, if the filter is canceled (Cancel button pressed) the store shall not be updated.

I understand. But why is `setApplyFilter` called?